kore/src/Kore/Builtin/Int.hs

```diff
@@ -510,9 +510,13 @@ unifyIntEq unifyChildren (NotSimplifier notSimplifier) unifyData =
     do
         solution <- unifyChildren operand1 operand2 & OrPattern.gather
         let solution' = MultiOr.map eraseTerm solution
-        (if value then pure else notSimplifier SideCondition.top) solution'
-            >>= Unify.scatter
+        scattered <-
+            Unify.scatter =<< case Bool.matchBool boolTerm of
+                Just False -> notSimplifier SideCondition.top solution'
+                _ -> pure solution'
```

Comment:
This is not correct. `boolTerm` could be `False`, or `True`, or it might not even be a `Bool` at all. Please make `boolTerm` be an [`InternalBool`](https://github.com/kframework/kore/blob/221cb82050490c6b711e1e68b42390971e60685a/kore/src/Kore/Internal/InternalBool.hs#L26) type, as I suggested in my [last review](https://github.com/kframework/kore/pull/2647/files/362e825afe15613934de7a4a88dec283e29499be#r656565430). That way, `unifyIntEq` is correctly defined over its inputs.